--- telemetrics-client-1.10.1/src/probes/oops_parser.c~	2016-06-30 23:41:26.000000000 +0000
+++ telemetrics-client-1.10.1/src/probes/oops_parser.c	2017-01-28 16:02:41.892028084 +0000
@@ -477,13 +477,6 @@
                 start++;
         }
 
-        if (*start && *start == '?') {
-                start++;
-                if (*start && isspace(*start)) {
-                        start++;
-                }
-        }
-
         offset_ptr = strchr(start, '+');
         end = offset_ptr ? offset_ptr : (start + strlen(start));
         frame->function = strndup(start, (size_t)(end - start));
