--- telemetrics-client-1.11.2/src/telemdaemon.c~	2017-01-19 21:24:12.000000000 +0000
+++ telemetrics-client-1.11.2/src/telemdaemon.c	2017-06-10 14:40:29.902986952 +0000
@@ -462,6 +462,8 @@
         const char *cert_file = get_cainfo_config();
         const char *tid_header = get_tidheader_config();
 
+	curl_global_init(CURL_GLOBAL_ALL);
+
         curl = curl_easy_init();
         if (!curl) {
                 telem_log(LOG_ERR, "curl_easy_init(): Unable to start libcurl"
@@ -532,6 +534,8 @@
 
         curl_slist_free_all(custom_headers);
         curl_easy_cleanup(curl);
+
+	curl_global_cleanup();
         return res ? false : true;
 }
 
