temporary patch: sleep longer waiting for debug info

to be upstreamed once field data shows it works

--- telemetrics-client-1.16.0/src/probes/crash_probe.c~	2018-03-09 22:00:27.000000000 +0000
+++ telemetrics-client-1.16.0/src/probes/crash_probe.c	2018-04-15 16:58:31.485003683 +0000
@@ -650,8 +650,8 @@
          * the first run (indicated by the presence of "??? - ["), wait 10
          * seconds and try again.
          */
-        if (strstr(backtrace->str, "??? - [")) {
-                sleep(10);
+        if (strstr(backtrace->str, "???")) {
+                sleep(30);
 
                 if (prepare_corefile(&e_core, core_fd) < 0) {
                         goto fail;
